dspace-api/src/main/java/org/dspace/versioning/VersionHistoryServiceImpl.java

```diff
@@ -81,6 +95,11 @@ public Version getVersion(VersionHistory versionHistory, Item item) {
        List<Version> versions = versionHistory.getVersions();
        for(Version v : versions)
        {
+           // skip soft-deleted versions
+           if (v.getItem() == null)
```

Comment:
I think we might need to replace the contents of this method with a call to a DAO that executes a database query. Might be more optimized instead of loading all versions into memory to return just one.